piet-direct2d/src/lib.rs

```diff
@@ -456,21 +457,43 @@ impl<'a> RenderContext for D2DRenderContext<'a> {
     fn capture_image_area(&mut self, rect: impl Into<Rect>) -> Result<Self::Image, Error> {
         let r = rect.into();
 
+        let (dpi_scale, _) = self.rt.get_dpi_scale();
+        let dpi_scale = dpi_scale as f64;
+
+        let transform_matrix = self.rt.get_transform();
+        let affine_transform = matrix3x2f_to_affine(transform_matrix);
+
+        let device_size = Point {
+            x: r.width() * dpi_scale,
+            y: r.height() * dpi_scale,
+        };
+        // TODO: This transformation is untested with the current test pictures
+        let device_size = affine_transform * device_size;
```

Comment:
I understand that `device_size` is a `Point` because only `Point` implements the `Affine` multiplication. However as soon as we're done with that, let's convert it to an actual `Size` as it makes subsequent code more robust. Add the following line after this:

```rust
let device_size = device_size.to_vec2().to_size();
```